Interfaces/IArguments.cs

```diff
@@ -0,0 +1,14 @@
+﻿using MyConsoleApp.UniqueFilenameClass;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyConsoleApp.Interfaces
+{
+    public interface IArguments
+    {
+        List<ArgumentsDto> ParseArguments(string[] args);
```

Comment:
Along the same "good names" theme, we can be a little more descriptive. "Parse" is excellent, tells me what it is going to do. But what does it parse? And what does it give back? Recall in Clean Code that the name should almost be self documenting. So if this were me, I would make it something like "ParseCmdLineArgsToSrvNmAppIdDTOs" - It is long but it describes what is going to happen.